discord-library/src/main/java/com/blank/humanity/discordbot/utils/menu/DiscordMenuBuilder.java

```diff
@@ -0,0 +1,84 @@
+package com.blank.humanity.discordbot.utils.menu;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import net.dv8tion.jda.api.entities.Emoji;
+import net.dv8tion.jda.api.events.Event;
+import net.dv8tion.jda.internal.utils.tuple.Pair;
+
+public abstract class DiscordMenuBuilder<M extends DiscordMenu> {
+
+    public interface DiscordMenuActionWrapper<E extends Event, M extends DiscordMenu> {
+        public boolean wrap(E event, M menu, String argument);
+    }
+
+    private Map<Class<?>, List<DiscordMenuActionWrapper<?, M>>> wrappers = new HashMap<>();
```

Comment:
That's some generic code you've got there.

Reply:
Yeah, currently only supports the new ComponentMenu, but I wanted to be able to readd the ReactionMenu again too.